master/internal/task/allocation.go

```diff
@@ -325,6 +326,10 @@ func (a *Allocation) Receive(ctx *actor.Context) error {
 			return nil
 		}
 		if err := a.preemption.ReceiveMsg(ctx); err != nil {
+			if _, ok := err.(ErrAllocationUnfulfilled); ok {
```

Comment:
This is a race condition that occurs when the trial running code sends ```AllocationPreemptionSignal``` before the allocation is fully restored

This same race also causes ```AllocationRendezvousInfo``` to fail which trial code doesn't seem to retry the API call like it does for ```AllocationPreemptionSignal```.

Still unsure the best way to handle this race, ideally more code could be synchronous on the restore path but it is tough with different resources managers using the shared allocation code